Public/Src/Cache/ContentStore/Interfaces/Stores/IContentStore.cs

```diff
@@ -28,6 +29,11 @@ public interface IContentStore : IStartupShutdown
         ///     Gets a current stats snapshot.
         /// </summary>
         Task<GetStatsResult> GetStatsAsync(Context context);
+
+        /// <summary>
+        ///     Remove given content from all sessions.
+        /// </summary>
+        Task<DeleteResult> DeleteAsync(Context context, ContentHash contentHash);
     }

```

Comment:
I had assumed this would go on Session rather that Store, but one could argue that it should count as a session-less operation and should therefore go here. Are you making that argument or did it end up here by accident?

Reply:
Yes, I am making this argument. If our intent is to remove content from a _machine_, then our most base implementation should not be for a session.